protos/configuration.proto

```diff
@@ -105,6 +105,11 @@ message RaftConfig {
 
   // Take a snapshot when cumulative data since last snapshot exceeds a certain size in bytes.
   uint64 snapshot_interval_size = 5;
+
+  // The maximal Raft ID in the cluster, during the entire lifetime of the cluster.
+  // This is computed internally by the server and is read only for clients.
+  // The Raft ID of a new peer added to the cluster mast be higher than this value.
+  uint64 max_raft_id = 6;
```

Comment:
If this is an internal value, it is better to manage internally without exposing it to the user.

Reply:
The `ConsensusConfig.RaftConfig.MaxRaftID` value is used my the admin to set a unique number, say `ConsensusConfig.RaftConfig.MaxRaftID+1`. It will be updated automatically if the config-tx commits as valid.
The next time the admin will read it, it will be updated to `ConsensusConfig.RaftConfig.MaxRaftID+1`. 
This is a simpler and easier  approach than the one in fabric. In the SDK, the code for adding a node will check that the chosen RaftID for a new peer is indeed >`ConsensusConfig.RaftConfig.MaxRaftID`, or maybe even will populate it automatically by setting it to `ConsensusConfig.RaftConfig.MaxRaftID+1`, we'll see.

Reply:
PS our approach is more like what etcd does, where the admin sets the RaftIDs, with the added benefit and safety of recording the max-id ever used, and enforcing new peers to respect that.